File: maven-root/mtsa/src/main/java/RobotProtocol/src/main/java/robotProtocol/messages/FollowLineMessage.cpp
/*
 * FollowLineMessage.cpp
 *
 *  Created on: Apr 16, 2013
 *      Author: mariano
 */

#include "FollowLineMessage.h"
#include <stdlib.h>
#include <stdio.h>

namespace RobotProtocol {

FollowLineMessage FollowLineMessage::staticMessage = FollowLineMessage(0,0);

Message *FollowLineMessage::deserialize(char *serializedMessage){
	FollowLineMessage::staticMessage.initialize(serializedMessage);
	return &(FollowLineMessage::staticMessage);
}

FollowLineMessage::FollowLineMessage(int to, int from)
:Message(MESSAGE_ID, 1, to, from){}

FollowLineMessage::FollowLineMessage(char *serializedMessage){
	initialize(serializedMessage);
}

FollowLineMessage::~FollowLineMessage() {
}

int FollowLineMessage::getMessageLength(){
	return Message::getMessageLength();
}

void FollowLineMessage::readData(char **serializedMessageIndex){}

void FollowLineMessage::writeData(char **serializedMessageIndex){}

} /* namespace RobotProtocol */

File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/FollowLineMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class FollowLineMessage extends Message {	
	public static int MESSAGE_ID				= 202;
	
	public FollowLineMessage(int to, int from){
		super(MESSAGE_ID, 1, to, from);
	}
	
	public FollowLineMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength();
	}
	
	
	@Override
	protected void readData(String inputString){
	}

	@Override
	protected String writeData(){
		return "";
	}

	@Override
	protected boolean hasReply() {
		return true;
	}
	
}
